machine/v1beta1/types_vsphereprovider.go

```diff
@@ -198,8 +198,34 @@ type VSphereDisk struct {
 	// +kubebuilder:validation:Maximum=16384
 	// +required
 	SizeGiB int32 `json:"sizeGiB"`
+	// provisioningMode is an optional field that specifies the provisioning type to be used by this vSphere data disk.
+	// Allowed values are "Thin", "Thick", "EagerlyZeroed", and omitted.
+	// When set to Thin, the disk will be made using thin provisioning allocating the bare minimum space.
+	// When set to Thick, the full disk size will be allocated when disk is created.
+	// When set to EagerlyZeroed, the full disk size will be allocated and all bytes will be zeroed.
+	// When omitted, no setting will be applied to the data disk and the provisioning mode for the disk will be provided by the default storage policy.
+	// +optional
+	ProvisioningMode ProvisioningMode `json:"provisioningMode,omitempty"`
 }
 
+// provisioningMode represents the various provisioning types available to a VMs disk.
+// +kubebuilder:validation:Enum=Thin;Thick;EagerlyZeroed
+type ProvisioningMode string
+
+const (
+	// ProvisioningModeThin creates the disk using thin provisioning. This means a sparse (allocate on demand)
+	// format with additional space optimizations.
+	ProvisioningModeThin ProvisioningMode = "Thin"
+
+	// ProvisioningModeThick creates the disk with all space allocated.
+	ProvisioningModeThick ProvisioningMode = "Thick"
+
+	// ProvisioningModeEagerlyZeroed creates the disk using eager zero provisioning. An eager zeroed thick disk
+	// has all space allocated and wiped clean of any previous contents on the physical media at
+	// creation time. Such disks may take longer time during creation compared to other disk formats.
```

Comment:
This description of the `EagerlyZeroed` provisioning mode is much more descriptive than the one that would currently be shown to users. I would recommend taking this description and using it in the actual field description where you say "When set to EagerlyZeroed ...".

Particularly the part about this provisioning mode taking longer during creation compared to the other disk formats would be helpful information for users to know when exploring this API.

Reply:
Ok, i can update that.